scripts/gen_app_partitions.py

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env python3
+import sys
+import argparse
+import os
+import re
+import string
+from elf_helper import ElfHelper
+from elftools.elf.elffile import ELFFile
+
+
+# This script will create linker comands for power of two aligned MPU
+# when APP_SHARED_MEM is enabled.
+print_template = """
+/* Auto generated code do not modify */
+. = ALIGN( 1 << LOG2CEIL(data_smem_REPLACE_STRINGb_end - data_smem_REPLACE_STRING));
```

Comment:
I believe it is better to use {0} and "".format() than REPLACE_STRING and .replace().